src/SelfService/Infrastructure/BackgroundJobs/UpdateOutOfSyncECRRepos.cs

```diff
@@ -0,0 +1,52 @@
+using SelfService.Domain.Services;
+using SelfService.Infrastructure.Api.Metrics;
+
+namespace SelfService.Infrastructure.BackgroundJobs;
+
+public class UpdateOutOfSyncEcrRepos : BackgroundService
+{
+    private readonly IServiceProvider _serviceProvider;
+    private readonly ILogger<UpdateOutOfSyncEcrRepos> _logger;
+
+    public UpdateOutOfSyncEcrRepos(IServiceProvider serviceProvider, ILogger<UpdateOutOfSyncEcrRepos> logger)
+    {
+        _serviceProvider = serviceProvider;
+        _logger = logger;
+    }
+
+    protected override Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        return Task.Run(
+            async () =>
+            {
+                while (!stoppingToken.IsCancellationRequested)
+                {
+                    await DoWork(stoppingToken);
+                    await Task.Delay(TimeSpan.FromMinutes(1), stoppingToken); //maybe each minute is a lot?
```

Comment:
I would make it lazy instead. Just update on need if its older than some amount of time (12 hours?)